sdcm/sct_runner.py

```diff
@@ -62,185 +87,285 @@ def instance_root_disk_size(test_duration) -> int:
+    @abstractmethod
+    def _get_instance_id(self, instance: Any) -> Any:
+        ...
+
+    @abstractmethod
+    def get_instance_public_ip(self, instance: Any) -> str:
+        ...
+
+    @abstractmethod
+    def _create_image(self, instance: Any) -> Any:
+        ...
+
+    @abstractmethod
+    def _get_image_id(self, image: Any) -> Any:
+        ...
+
+    @abstractmethod
+    def _copy_source_image_to_region(self) -> None:
+        ...
+
+    def create_image(self) -> None:
+        """Create SCT Runner image in specified region.
+
+        If the image exists in SOURCE_REGION it will be copied to the destination region.
+
+        WARNING: this can't run in parallel!
         """
+        LOGGER.info("Looking for source SCT Runner Image in `%s'...", self.SOURCE_IMAGE_REGION)
+        source_image = self.source_image
+        if not source_image:
+            LOGGER.info("Source SCT Runner Image not found. Creating...")
+            instance = self._create_instance(
+                instance_type=self.IMAGE_BUILDER_INSTANCE_TYPE,
+                base_image=self.BASE_IMAGE,
+                tags={
+                    "keep": "1",
+                    "keep_action": "terminate",
+                    "Version": self.VERSION,
+                },
+                instance_name=f"{self.image_name}-builder-{self.SOURCE_IMAGE_REGION}",
+                region_az=self.region_az(
+                    region_name=self.SOURCE_IMAGE_REGION,
+                    availability_zone=self.availability_zone,
+                ),
+            )
+            self.install_prereqs(public_ip=self.get_instance_public_ip(instance=instance), connect_timeout=120)
+
+            LOGGER.info("Stopping the SCT Image Builder instance...")
+            self._stop_image_builder_instance(instance=instance)
+
+            LOGGER.info("SCT Image Builder instance stopped.\nCreating image...")
+            self._create_image(instance=instance)
+
+            builder_instance_id = self._get_instance_id(instance=instance)
+            try:
+                LOGGER.info("Terminating SCT Image Builder instance `%s'...", builder_instance_id)
+                self._terminate_image_builder_instance(instance=instance)
+            except Exception as ex:  # pylint: disable=broad-except
+                LOGGER.warning("Was not able to terminate `%s': %s\nPlease terminate manually!!!",
+                               builder_instance_id, ex)
+        else:
+            LOGGER.info("SCT Runner image exists in the source region `%s'! ID: %s",
+                        self.SOURCE_IMAGE_REGION, self._get_image_id(image=source_image))
+
+        if self.region_name != self.SOURCE_IMAGE_REGION and self.image is None:
+            LOGGER.info("Copying %s to %s...\nNOTE: it can take 5-15 minutes.",
+                        self.image_name, self.region_name)
+            self._copy_source_image_to_region()
+            LOGGER.info("Done.")
```

Comment:
Between "start of copying" message and this one something else may write other messages. So, I guess it is better to add more details to this particular message than just "Done". For example:

```
LOGGER.info(
    "Finished copying %s to %s...\nNOTE: it can take 5-15 minutes.",
    self.image_name, self.region_name)
```